tests/lib/tools/log-parser

```diff
@@ -0,0 +1,545 @@
+#!/usr/bin/env python3
+
+"""
+This tool reads a spread log and creates a file with all the data
+The output file includes the more importan information extracted
+from the log to be analyzed
+"""
+
+import argparse
+import json
+import os
+import re
+import sys
+
+# Info types
+ERROR_TYPE = 'Error'
+DEBUG_TYPE = 'Debug'
+WARN_TYPE = 'WARNING:'
+
+# Results
+FAILED_TYPE = 'Failed'
+ABORTED_TYPE = 'Aborted'
+SUCCESSFUL_TYPE = 'Successful'
+
+# Printable names
+ALL = 'all'
+NONE = 'none'
+ACTION = 'action'
+OPERATION = 'operation'
+INFO = 'info'
+ERROR = 'error'
+ERROR_DEBUG = 'error-debug'
+FAILED = 'failed'
+ABORTED = 'aborted'
+SUCCESSFUL = 'successful'
+
+RESULT = 'result'
+START = 'Found'
+SPREAD_FILE = 'spread.yaml'
+
+EXEC_VERBS = ['Preparing', 'Executing', 'Restoring']
+INFO_TYPES = [ERROR_TYPE, DEBUG_TYPE, WARN_TYPE]
+OPERATIONS = [
+    'Rebooting', 'Discarding', 'Allocating', 'Waiting',
+    'Allocated', 'Connecting', 'Connected', 'Sending'
+    ]
+RESULTS = ['Successful', 'Aborted', 'Failed']
+
+
+class Action:
+    """
+    Action represents the main spread tasks actions
+    The actions can be: Preparing, Executing and Restoring
+    """
+
+    def __init__(self, verb, task, date, time, source_line):
+        self.type = ACTION
+        self.verb = verb
+        self.time = time
+        self.date = date
+        self.task = task
+        self.source_line = source_line
+
+    def __repr__(self):
+        return self.source_line
+
+    def __dict__(self):
```

Comment:
Why do we have all these `__dict__()` methods? What's the difference compared to using the default ones?

Reply:
It is because when I print (save to file) the objects I need to just print some data, and in __dict__() I change the data I need to be printed.  I'll add a comment in the function explaining this. 
Thanks for the review